### PR TITLE
feat: Update docs about recent changes related to kyma-dashboard

### PR DESCRIPTION
**Pull request by dbadura**
**Description**

Changes proposed in this pull request:

- Adjust docs to new configuration management
- 

**Related issue(s)**
https://github.com/kyma-project/busola/issues/3298

**Definition of done**

- [x] The PR's title starts with one of the following prefixes:
  - feat: A new feature
  - fix: A bug fix
  - docs: Documentation only changes
  - refactor: A code change that neither fixes a bug nor adds a feature
  - test: Adding tests
  - chore: Maintainance changes to the build process or auxiliary tools, libraries, workflows, etc.
- [x] Related issues are linked. To link internal trackers, use the issue IDs like `backlog#4567`
- [x] Explain clearly why you created the PR and what changes it introduces
- [x] All necessary steps are delivered, for example, tests, documentation, merging

### CODE REVIEW

File: kyma/README.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
+> **NOTE:** This is a general template that you can use for a project README.md. Except for the mandatory sections, use only those sections that suit your use case but keep the proposed section order.
+>
+> Mandatory sections:
+> - `Overview`
+> - `Prerequisites`, if there are any requirements regarding hard- or software
+> - `Installation`
+> - `Contributing` - do not change this!
+> - `Code of Conduct` - do not change this!
+> - `Licensing` - do not change this!
```
Comment (mrCherry97) [inline kyma/README.md:9]:
```suggestion
```

File: kyma/README.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+The **environments** directory contains configuration per environment. 
+The environment configuration contains **config.yaml** for busola and **extensions.json**, **statics.json**, **wizards.json** with list of extensibility configuration.
+The entries for extensibility can be **relative path to kyma** directory or **URL to extension yaml**.
+
+The **extensions** directory contains general extensibility configuration to avoid duplication in each environment configuration dir.
+
+## Prerequisites
+
+- [`npm`](https://www.npmjs.com/) in version 8.1.2 or higher
+- [`node`](https://nodejs.org/en/) in version 16.13.2 or higher
```
Comment (mrCherry97) [inline kyma/README.md:49]: Here we should have the same versions as for busola

**Review by mrCherry97 — CHANGES_REQUESTED** [review]

**Review by mrCherry97 — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -84,6 +86,34 @@ Features comprise the following elements:
 
 See the available Busola [feature flags](docs/features.md) for more information.
 
+#### Environment specific settings
```
Comment (IwonaLanger) [inline README.md:89]:
```suggestion
#### Environment-Specific Settings
```

File: README.md [hunk 4]
```diff
@@ -84,6 +86,34 @@ Features comprise the following elements:
...
+
+You can provide override to the default configuration with your own environment specific settings.
```
Comment (IwonaLanger) [inline README.md:91]:
```suggestion
You can provide an override to the default configuration with your own environment-specific settings.
```

File: README.md [hunk 5]
```diff
@@ -84,6 +86,34 @@ Features comprise the following elements:
...
+Custom environment directory structure should look and be placed in **public/environments**.
```
Comment (IwonaLanger) [inline README.md:92]:
```suggestion
Follow this pattern to structure your custom environment directory and place it in `public/environments`.
```

File: README.md [hunk 6]
```diff
@@ -84,6 +86,34 @@ Features comprise the following elements:
...
+```
+custom-env/
+├── config
+│   └── config.yaml
+└── extensions
+    ├── extensions.yaml
+    └── wizards.yaml
+```
+
+> **Important:** Files `extensions.yaml`, `statics.yaml`, `wizards.yaml` and `config.yaml` are required to make busola work properly.
```
Comment (IwonaLanger) [inline README.md:103]:
```suggestion
> [!WARNING]
> The `extensions.yaml`, `statics.yaml`, `wizards.yaml`, and `config.yaml` files are necessary for Busola to work properly.
```

File: README.md [hunk 7]
```diff
@@ -84,6 +86,34 @@ Features comprise the following elements:
...
+
+To activate environment configuration you have to create or edit `active.env` file located in [public](./public).
```
Comment (IwonaLanger) [inline README.md:105]:
```suggestion
To activate your environment configuration, create or edit the `active.env` file in the [public directory](./public).
```

File: README.md [hunk 8]
```diff
@@ -84,6 +86,34 @@ Features comprise the following elements:
...
+The `active.env` file should look like that:
```
Comment (IwonaLanger) [inline README.md:106]:
```suggestion
Follow this example of the `active.env` file:
```

File: README.md [hunk 9]
```diff
@@ -84,6 +86,34 @@ Features comprise the following elements:
...
+
+```dotenv
+ENVIRONMENT=your-environment-name
+```
+
+When the `ENVIRONMENT` is set to `my-env`, the busola looks for custom configuration in **public/environemnt/my-env**
```
Comment (IwonaLanger) [inline README.md:112]:
```suggestion
When **ENVIRONMENT** is set to `my-env`, Busola looks for your custom configuration in `public/environemnt/my-env`.
```

File: README.md [hunk 10]
```diff
@@ -84,6 +86,34 @@ Features comprise the following elements:
...
+If the `ENVIRONMENT` is not set busola fetch default configuration which has the same structure as custom configuration located in [public](./public)
```
Comment (IwonaLanger) [inline README.md:113]:
```suggestion
If **ENVIRONMENT** is not set, Busola fetches the default configuration with the same structure as the custom configuration located in the [public directory](./public).
```

File: README.md [hunk 11]
```diff
@@ -84,6 +86,34 @@ Features comprise the following elements:
...
+
+In case of docker image, the file `active.env` is created from `ENVIRONMENT` env at the startup to image.
```
Comment (IwonaLanger) [inline README.md:115]:
```suggestion
In the case of the Docker image, the `active.env` file is created from the **ENVIRONMENT** env at the startup of the image.
```

Comment (IwonaLanger) [reply README.md:115]: Again, I'm not sure if I understand it correctly. What is "env" here? If it's environment, is it necessary to say "ENVIRONMENT environment"?

File: kyma/README.md [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+The **extensions** directory contains general extensibility configuration to avoid duplication in each environment configuration dir.
+
+## Prerequisites
+
+- [`npm`](https://www.npmjs.com/) in version 10.x
+- [`node`](https://nodejs.org/en/) in version 20.x
+- [`make`](https://www.gnu.org/software/make/)
+
+## Installation
+
```
[the remark this answers is not in the document]
Comment (dbadura) [reply kyma/README.md:43]: It's used to download all dependencies needed to `make prepare-all-configuration` to work.

File: README.md (hunk 11 above)
Comment (dbadura) [reply README.md:116]: env is a shortcut of environment variable which is represented by `KEY=value`, so to make it more readable I used `env`. Additionaly it's quite common to name it like that.

File: kyma/README.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (IwonaLanger) [reply kyma/README.md:43]: I suggest adding the information here.

File: README.md (hunk 11 above)
Comment (IwonaLanger) [reply README.md:116]: If it's not the actual value, I recommend rephrasing it to:
In the case of the Docker image, the `active.env` file is created at the startup of the image from the environment specified in the **ENVIRONMENT** variable.

**Review by IwonaLanger — APPROVED** [review]

**Review by mrCherry97 — APPROVED** [review]

**Pull request merged**